Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+        }
+    }
+    
+    /*
+     A Boolean value that indicates whether the minimum and full extent
+     lables are visible or not. Default if True.
+     */
+    public var fullExtentLabelsVisible = true {
+        didSet {
+            fullExtentStartTimeLabel.isHidden = !fullExtentLabelsVisible
+            fullExtentEndTimeLabel.isHidden = !fullExtentLabelsVisible
+            if fullExtentLabelsVisible {
+                updateFullExtentLabelFrames()
+            }
+        }
+    }
+    
+    /*
+     The date formatter used for full extent dates.
+     */
+    public var fullExtentDateFormatter: DateFormatter = {
+        let formatter: DateFormatter = DateFormatter()
+        formatter.locale = Locale.current
+        formatter.timeZone = TimeZone.current
+        formatter.dateStyle = .medium
+        return formatter
+    }()
+    
+    // MARK: Time Step Properties
+    
+    /*
+     The time steps calculated along the time slider. These are
+     calculated based on the full extent and time step interval.
+     */
+    public private(set) var timeSteps: [Date]?
+    
+    /*
+     The interval at which users can move forward and
+     back through the slider's full extent.
+     */
+    public var timeStepInterval: AGSTimeValue? {
+        didSet {
+            calculateTimeSteps()
+            refresh()
+        }
+    }
+    
+    /*
+     The date formatter used for time interval or tick mark dates.
+     */
+    public var timeStepIntervalDateFormatter: DateFormatter = {
+        let formatter: DateFormatter = DateFormatter()
+        formatter.locale = Locale.current
+        formatter.timeZone = TimeZone.current
+        formatter.dateStyle = .medium
+        return formatter
+    }()
+    
+    /*
+     The time interval or tick mark label color.
+     */
+    public var timeStepIntervalLabelColor: UIColor = UIColor.darkText {
+        didSet {
+            tickMarkLabels.forEach { (tickMarkLabel) in
+                tickMarkLabel.foregroundColor = timeStepIntervalLabelColor.cgColor
+            }
+        }
+    }
+    
+    /*
+     The current extent label font.
+     */
+    public var timeStepIntervalLabelFont: UIFont = UIFont.systemFont(ofSize: 12.0) {
+        didSet {
+            tickMarkLabels.forEach { (tickMarkLabel) in
+                tickMarkLabel.font = timeStepIntervalLabelFont as CFTypeRef
+                tickMarkLabel.fontSize = timeStepIntervalLabelFont.pointSize
+            }
+        }
+    }
+    
+    /*
+     The color used to tint the full extent of the slider.
+     */
+    public var timeStepIntervalTickColor: UIColor = UIColor.black {
+        didSet {
+            tickMarkLayer.setNeedsDisplay()
+        }
+    }
+    
+    /*
+     Controls how labels on the slider are displayed. The default is thumbs.
+     */
+    public var labelMode: TimeSliderLabelMode = .thumbs {
+        didSet {
+            switch labelMode {
```

Comment:
It looks like the content of the first and third cases are identical, so they could be combined using `case .none, .ticks:`.

Reply:
fixed